Album add ons

cc13985 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * In-card album download with visual loading and toasts; bulk selection toolbar for selecting, downloading, and deleting items.

* **Documentation**
  * Added comprehensive developer documentation covering architecture, setup, data model, and known issues.

* **Style**
  * Updated album card and gallery UI to use Mantine components for improved consistency and layout and selectable item groups.

* **Chores**
  * Added jszip dependency; removed react-router-dom; allowed placeholder remote images.

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/Hack4Impact-UMD/camp-starfish/pull/23)
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cc13985 committed:
Added multiselect checkboxes

cc13985 committed:
Updated sizing of checkboxes

cc13985 committed:
Added gray background div and adjusted sizing

cc13985 committed:
Added download button

cc13985 committed:
Made icons show on hover

cc13985 committed:
Added showControls check to have sticky icons

cc13985 committed:
Added admin/photographer view for album cards

nkanchinadam commented on src/components/AlbumCard.tsx:
```diff
@@ -1,25 +1,186 @@
-import React from "react";
+import React, { useRef, useEffect } from "react";
```
You can just import useState directly instead of calling React.useState for state.

nkanchinadam reviewed (changes requested):
It looks good, but the implementation directly manipulates the DOM, which is bad practice in React. Additionally, the component needs to be merged with the current version in main for type consistency.

nkanchinadam commented:
Currently, the AlbumCard component displays information about an Album. This PR is to add the options that appear on the AlbumCard when a user hovers over it as shown in this [Figma frame](https://www.figma.com/design/oT2K7Ph5FnsooXHkYzkrYL/Photo-Management-Tool?node-id=3310-20795&m=dev)

Before we can merge this PR in, we need to do the following:
- [ ] Merge the latest changes from main into this branch
- [ ] Update the UI elements to use their respective prebuilt Mantine components
- [ ] Implement download functionality for downloading an entire album

Chase-Fournier committed:
Merge branch 'main' into album-add-ons

Chase-Fournier committed:
Finishing updates, adding downloads

Chase-Fournier committed:
Get rid of Orphen

Chase-Fournier committed:
Update Claude.md

Chase-Fournier committed:
Fixes to issues

cc13985 committed:
feat(albums): improve AlbumPage UI/UX & implement album upload/download functionality (#28)

- Improved AlbumPage UI/UX
  - Integrated with Mantine & Tanstack Query
  - Replaced file icons with Material Design icon components from the `react-icons` library
  - Options pane includes tag multi-select, link to pending page, upload icon, and download icon
- Implemented album item uploading
  - Through the `UploadAlbumItemsModal`, users can upload album items
  - Each album item's state is handled independently since partial success when uploading a large amount of album items is possible
  - Album items are automatically set as in review or approved based on the uploading user's role
    - Roles without permissions to upload album items see an error page in the modal body
- Implemented album downloading
  - Users with permission to download album items can do so by clicking the download icon
  - Download functionality takes in Firestore query filters to only allow users to download the album items that they are authorized to view

Co-authored-by: Nitin Kanchinadam <nitin.kanchinadam@gmail.com>

Chase-Fournier committed:
Small code rabbit changes

Chase-Fournier committed:
smail change

Chase-Fournier committed:
Merge branch 'main' into album-add-ons

Chase-Fournier committed:
Lint errors

Chase-Fournier committed:
Small Fixes

Chase-Fournier committed:
Tool Bar

Chase-Fournier committed:
Merge branch 'main' into album-add-ons

Chase-Fournier committed:
Merge branch 'main' into album-add-ons
